src/lib/modes.js

```diff
@@ -73,12 +73,25 @@ export const COMMENT = function(begin, end, modeOptions = {}) {
     },
     modeOptions
   );
-  mode.contains.push(PHRASAL_WORDS_MODE);
+  // mode.contains.push(PHRASAL_WORDS_MODE);
   mode.contains.push({
     className: 'doctag',
-    begin: '(?:TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):',
+    // hack to avoid the space from being included. the space is necessary to
+    // match here to prevent the plain text rule below from gobbling up doctags
+    begin: '[ ]*(?=(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):)',
+    end: /(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):/,
```

Comment:
```suggestion
    begin: '[ ]*(?=(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):?)',
    end: /(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):?/,
```

I've seen code snippets that don't have the `:`, so unless this causes problems, I'd prefer the `:` be optional.

Reply:
Please provide some examples?... opening this up makes this a LOT fuzzier... perhaps we need a word boundary then... or else you match NOTEABLE, XXXXXXXX, BUGGY, BUGGER, etc... what does it look like when someone doesn't use a `:`?  I kind of like the precision of it as it stands now... that `:` is the "one true way". LOL.

Reply:
You're right, a word boundary would be better in this case. Definitely not as common to not have a `:`, but code linters often support it, e.g.

- https://github.com/JohnPostlethwait/fixme/blob/master/bin/fixme.js
- https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/Generic/Tests/Commenting/FixmeUnitTest.js
- https://github.com/matteosan1/workspace/blob/master/qt4/tdpalio2online/tdpalioserver/corsa.cpp

However, I don't feel strongly about this